Work around WKTR installation issue

pvollan opened the pull request:
#### f3ee3389d13287450b02e9c789d00b62562789b4
<pre>
Work around WKTR installation issue
<a href="https://bugs.webkit.org/show_bug.cgi?id=275537">https://bugs.webkit.org/show_bug.cgi?id=275537</a>
<a href="https://rdar.apple.com/129928587">rdar://129928587</a>

Reviewed by Alexey Proskuryakov.

When embedding WebKit process extensions in the WebKitTestRunner app bundle, the Simuator installation
sometimes fail. This is caused by a race condition where the directory /private/var/db is not always
created when a system service taking part in the installation process is starting up. This patch
addresses this issue by creating the directory before the Simulator instance is booted.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager._boot_device):

Canonical link: <a href="https://commits.webkit.org/280121@main">https://commits.webkit.org/280121@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/df0729b05aee3f35f961b78f31ad1278b3ab7b32

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55778 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35101 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8245 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58762 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6209 "Build is in progress. Recent messages:Printed configuration; Running apply-patch; Checked out pull request; Compiled WebKit") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57904 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42723 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6407 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44922 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4278 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57807 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33000 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48082 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26056 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/55306 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29786 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5411 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4352 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51755 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5680 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60353 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5809 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52349 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48153 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51846 "Passed tests") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8233 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30932 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32017 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33098 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31764 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
```diff
@@ -367,6 +367,10 @@ def _wait_until_device_is_usable(device, deadline):
         host = host or SystemHost.get_default()
+
+        # FIXME: remove this workaround after rdar://129789675 has been resolved.
+        host.executive.run_command(['sh', '-c', "mkdir -m 755 -p " + "~/Library/Developer/CoreSimulator/Devices/" + device.udid + "/data/private/var/db"])
```
Are these bits correct? One I have on my machine is `700`:

```
drwxr-xr-x  3 ap  staff   96 Mar 27 16:25 MobileAsset
drwx------  5 ap  staff  160 Mar 27 16:25 db
drwx--x--x  3 ap  staff   96 Mar 27 16:25 protected
```

pvollan replied:
That's a good point, I think this can be changed to 700.

Thanks for reviewing!

pvollan committed:
Work around WKTR installation issue
https://bugs.webkit.org/show_bug.cgi?id=275537
rdar://129928587

Reviewed by Alexey Proskuryakov.

When embedding WebKit process extensions in the WebKitTestRunner app bundle, the Simuator installation
sometimes fail. This is caused by a race condition where the directory /private/var/db is not always
created when a system service taking part in the installation process is starting up. This patch
addresses this issue by creating the directory before the Simulator instance is booted.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager._boot_device):

Canonical link: https://commits.webkit.org/280121@main
